test/e2e/upgrade_management_cluster_test.go

```diff
@@ -175,18 +174,13 @@ func upgradeManagementCluster() {
 		Expect(err).To(BeNil())
 
 		By("Check if BMH is in provisioned state")
-		Eventually(func() error {
+		Eventually(func(g Gomega) {
 			bmhList := &bmov1alpha1.BareMetalHostList{}
-			if err := upgradeClusterClient.List(ctx, bmhList, client.InNamespace(namespace)); err != nil {
-				return err
-			}
+			g.Expect(upgradeClusterClient.List(ctx, bmhList, client.InNamespace(namespace))).Should(Succeed())
 			for _, bmh := range bmhList.Items {
-				if !bmh.WasProvisioned() {
-					return errors.New("BMHs cannot be provisioned")
-				}
+				g.Expect(bmh.WasProvisioned()).To(BeTrue())
 			}
-			return nil
-		}, e2eConfig.GetIntervals(specName, "wait-bmh-provisioned")...).Should(BeNil())
+		}, e2eConfig.GetIntervals(specName, "wait-bmh-provisioned")...).Should(Succeed())
 
 		Logf("Apply the available BMHs CRs")
 		cmd = exec.Command("kubectl", "apply", "-f", "/opt/metal3-dev-env/bmhosts_crs.yaml", "-n", namespace, "--kubeconfig", upgradeClusterProxy.GetKubeconfigPath())
```

Comment:
Can we change this code block too and make it like other Eventuallys: 
https://github.com/metal3-io/cluster-api-provider-metal3/blob/3482eab06e06b087454b0c94139aa1c4da58d6c5/test/e2e/upgrade_management_cluster_test.go#L244

Reply:
It is a bit different than the others because it is working with v1alpha4 Machines. I could change it to look more like the other but would rather do that in the follow up PR (#714)